ui/plugin-system/src/components/CalculationSelector/CalculationSelector.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { FormControl, TextField, Autocomplete } from '@mui/material';
+import { CALCULATIONS_CONFIG, CalculationConfig, CalculationType } from '../../model/calculations';
```

Comment:
Since CalculationSelector, CalculationsMap, CalculationType are used in multiple panels (2 of our 3 panels currently, but I don't see them being used in upcoming Table / BarChart panels), but aren't relevant for all panel plugins, I think putting them in @perses-dev/components might make more sense. This would be consistent with where units.ts and UnitSelector.tsx live currently. 

Another option is the types / interfaces go in @perses-dev/core and just CalculationSelector goes in components, which wouldn't create a circular dependency. I'm definitely open to feedback here, it isn't 100% clear to me what all should go in plugin-system, I mostly just want things to be consistent

Reply:
I'm really struggling with that too. I think we need some stronger "rule of thumbs" for what goes where. It feels a little random right now. E.g. the `UnitSelector` went into `components` because the `units` model was in there.  I'm going to start a small conversation about this in slack. 

If I recall correctly, for this particular item, I'll end up in a circular dep if the whole thing goes into `components`, but putting the types into `core` should fix that.

Reply:
We agreed to punt on reorganizing this in this PR because it will require some deeper refactoring of some other models depending on calculation. It will be revisited in the future.